### PR TITLE
Switch to Ruff

### PR DESCRIPTION
**Pull request by tuomas777**
This PR introduces **Ruff** as the new linter and formatter, replacing the previously used tools: **Black**, **Flake8**, and **isort**. The goal is also to standardize the linting and formatting rules across all of our backend projects.

One key change to linting/formatting rules is the reduction of the linter maximum line length from **120** to **88** characters. Consequently, hundreds of lines exceeding the new limit are marked with `# noqa: E501` to prevent linting errors. These comments can be removed gradually as the codebase is updated to comply with the new standard.

The PR includes a significant number of modified lines, primarily due to automatic formatting by Ruff. **Reviewers can safely skip the following commits during manual review, as they involve safe automatic formatting:**
- [0a0b506](https://github.com/City-of-Helsinki/linkedevents/pull/1005/commits/0a0b50668de376fb2b08ac681e49c67f35f74aa1)
- [0a16d94](https://github.com/City-of-Helsinki/linkedevents/pull/1005/commits/0a16d948eeb779baf68a8867a4608f90d4904909)

Some automatic changes made by Ruff and **autopep8** (which was run to help fix excessively long lines) may require some manual inspection, as they could be potentially unsafe:
- [46665ca](https://github.com/City-of-Helsinki/linkedevents/pull/1005/commits/46665ca57bd8791983f6e1f78d92ae82b6f47cd9)
- [4d206c8](https://github.com/City-of-Helsinki/linkedevents/pull/1005/commits/4d206c8a2a5f262b61789b49900206b89071986e)

After the review and before merging, the plan is to squash commits for a cleaner history, perhaps combining the current `chore` commits into one and the `style` commits into another (which can then be added to `.git-blame-ignore-revs`).

**Update**: Added some new rules to Ruff lint config:

- `flake8-bugbear` (excluding opinionated rules 9xx) and `pep8-naming`: these are in use with Flake8 in many of our projects, so let's keep using them
- `flake8-print` and `flake8-pie`: these are pretty trivial to add

New commits start from https://github.com/City-of-Helsinki/linkedevents/pull/1005/commits/cc322645eb99aab472caff776f9156e0577e00c2

Refs [LINK-2201](https://helsinkisolutionoffice.atlassian.net/browse/LINK-2201)

[LINK-2201]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-2201?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

File: requirements-dev.txt [hunk 1]
```diff
@@ -81,44 +66,25 @@ inflection==0.5.1
...
 iniconfig==2.0.0
     # via pytest
-isort==5.13.2
-    # via -r requirements-dev.in
 markupsafe==2.1.5
     # via
     #   -c requirements.txt
     #   werkzeug
-mccabe==0.7.0
-    # via flake8
```
Comment (danipran) [inline requirements-dev.txt:91]: Should [mccabe (C90)](https://docs.astral.sh/ruff/rules/#mccabe-c90) be included in linting rules as well?

**Review by danipran — APPROVED** [review]
LGTM, left a question about mccabe

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

File: requirements-dev.txt (hunk 1 above)
Comment (tuomas777) [reply requirements-dev.txt:91]: It was agreed that `mccabe` is unnecessary and will be left out.

**Review by danipran — APPROVED** [review]
LGTM! 👍

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**LINKEDEVENTS-API branch is deployed to platta: https://linkedevents-pr1005.api.dev.hel.ninja** :rocket::rocket::rocket:

**Pull request merged**